src/library/dao/cratedao.cpp

```diff
@@ -18,6 +18,32 @@ CrateDAO::~CrateDAO() {
 
 void CrateDAO::initialize() {
     qDebug() << "CrateDAO::initialize()";
+
+    //get the count to allocate HashMap
+    int tracksInCratesCount = 0;
+    QSqlQuery query(m_database);
+    query.prepare("SELECT COUNT(*) from " CRATE_TRACKS_TABLE);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+    }
+
+    tracksInCratesCount = query.value(0).toInt();
+
+    m_cratesTrackIsIn.reserve(tracksInCratesCount);
+
+    //now fetch all Tracks from all crates and insert them into the hashmap
+    query.prepare("SELECT track_id, crate_id from " CRATE_TRACKS_TABLE);
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+    }
+
+    const int trackIdColumn = query.record().indexOf("track_id");
+    const int crateIdColumn = query.record().indexOf("crate_id");
+    while (query.next()) {
+        m_cratesTrackIsIn.insert(query.value(crateIdColumn).toInt(),
+                                 query.value(trackIdColumn).toInt());
+    }
```

Comment:
This basically copies the whole crates table to memory. 
Up to now we have to care about this table and the original library table. 
This is extra work for all crate actions and a source of errors because of two data storages. 

Do we have test data, how this compares to the original solution?
I think we have a penalty on mass actions.
Is it worth the work? Remember that the selected Track slot is already decoupled by a timer.   

Is there an alternative way to speed up the sql lookups e.g. a temp table?